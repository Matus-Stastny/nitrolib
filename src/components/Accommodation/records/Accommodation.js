// @flow strict

import type { RoomType } from "../../../records/Room";

export type AccommodationType = {|
  hotelName: string,
  rating: number,
  location: {|
    latitude: number,
    longitude: number,
  |},
  address: string,
  photoUrl: string,
  rooms: Array<RoomType>,
  mapboxToken: string,
  zoom?: number,
  isMMB: boolean,
|};
