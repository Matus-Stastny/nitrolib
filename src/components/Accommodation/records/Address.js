// @flow strict

export type AddressType = ?{|
  +fullAddress: ?string,
|};
