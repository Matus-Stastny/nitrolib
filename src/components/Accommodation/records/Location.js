// @flow strict

export type LocationType = {|
  longitude: number,
  latitude: number,
  address: string,
  hotelName: string,
  mapboxToken: string,
  desktopWidth?: number,
  zoom?: number,
|};
