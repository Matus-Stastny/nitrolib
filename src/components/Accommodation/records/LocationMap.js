// @flow strict

export type LocationType = {|
  center: {|
    latitude: number,
    longitude: number,
  |},
  label: string,
  zoom: number,
  desktopWidth: number,
  mapboxToken: string,
|};
