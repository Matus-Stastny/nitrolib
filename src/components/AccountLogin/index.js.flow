// @flow strict
import * as React from "react";

type LoginType = "mmb" | "help" | "refer";

type Props = {|
  type?: LoginType,
  email: string,
  error?: React.Node,
  isLoading?: boolean,
  onGoogleLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onFacebookLogin: (ev: SyntheticEvent<HTMLButtonElement>) => void,
  onEmailChange: (ev: SyntheticInputEvent<HTMLInputElement>) => void,
  onContinue: (ev: SyntheticEvent<HTMLButtonElement>) => void,
|};

declare export default React.ComponentType<Props>;
