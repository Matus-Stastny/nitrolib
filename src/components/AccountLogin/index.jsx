// @flow strict
import * as React from "react";
import OrbitText from "@kiwicom/orbit-components/lib/Text";
import Heading from "@kiwicom/orbit-components/lib/Heading";
import InputField from "@kiwicom/orbit-components/lib/InputField";
import TextLink from "@kiwicom/orbit-components/lib/TextLink";
import Button from "@kiwicom/orbit-components/lib/Button";
import Illustration from "@kiwicom/orbit-components/lib/Illustration";
import FacebookIcon from "@kiwicom/orbit-components/lib/icons/Facebook";
import GoogleIcon from "@kiwicom/orbit-components/lib/icons/Google";
import Stack from "@kiwicom/orbit-components/lib/Stack";
import Section, { StyledModalSection } from "@kiwicom/orbit-components/lib/Modal/ModalSection";
import styled from "styled-components";

import Text from "../Text";
import { Consumer } from "../../services/intl/context";

// temporary fix for InputField width, will be fixed in orbit-components ASAP
// TODO: replace with fragment
const Wrapper = styled.div`
  ${StyledModalSection} label {
    width: 100%;
  }
`;

type IllustrationType = "Login" | "Help" | "InviteAFriend" | "AirHelp";

type Props = {|
  illustration?: IllustrationType,
  text?: string,
  email: string,
  onNoAccount: (ev?: SyntheticEvent<HTMLLinkElement>) => void,
  onGoogleLogin: (ev?: SyntheticEvent<HTMLButtonElement>) => void,
  onFacebookLogin: (ev?: SyntheticEvent<HTMLButtonElement>) => void,
  onEmailChange: (ev: SyntheticInputEvent<HTMLInputElement>) => void,
  onContinue: (ev?: SyntheticEvent<HTMLButtonElement>) => void,
|};

const AccountLogin = ({
  email,
  text,
  illustration,
  onNoAccount,
  onGoogleLogin,
  onFacebookLogin,
  onEmailChange,
  onContinue,
}: Props) => (
  <Consumer>
    {intl => (
      <Wrapper>
        <Section>
          <Illustration name={illustration || "Login"} size="small" spaceAfter="small" />
          <Heading element="h2" spaceAfter="small">
            {text || <Text t={__("account.manage_your_bookings")} />}
          </Heading>
          <OrbitText spaceAfter="large">
            <Text t={__("account.sign_in_description")} />
          </OrbitText>
        </Section>
        <Section>
          <OrbitText weight="bold" spaceAfter="medium">
            <Text t={__("account.sign_in_description")} />
          </OrbitText>
          <Stack
            spaceAfter="small"
            desktop={{
              spacing: "condensed",
              direction: "row",
              align: "end",
            }}
          >
            <InputField
              label={intl.translate(__("account.email"))}
              placeholder={intl.translate(__("account.email_placeholder"))}
              type="email"
              value={email}
              onChange={onEmailChange}
            />
            <Button onClick={onContinue}>
              <Text t={__("account.continue")} />
            </Button>
          </Stack>
          <OrbitText size="small">
            <TextLink type="secondary" onClick={onNoAccount}>
              <Text t={__("account.i_dont_have_account")} />
            </TextLink>
          </OrbitText>
        </Section>
        <Section suppressed>
          <OrbitText weight="bold" spaceAfter="medium">
            <Text t={__("account.or_social_account")} />
          </OrbitText>
          <Stack desktop={{ spacing: "natural", direction: "row", align: "end" }}>
            <Button
              type="facebook"
              block
              bordered
              icon={<FacebookIcon />}
              onClick={onFacebookLogin}
            >
              <Text t={__("account.log_in_with")} values={{ provider: "Facebook" }} />
            </Button>
            <Button type="google" block bordered icon={<GoogleIcon />} onClick={onGoogleLogin}>
              <Text t={__("account.log_in_with")} values={{ provider: "Google" }} />
            </Button>
          </Stack>
        </Section>
      </Wrapper>
    )}
  </Consumer>
);

export default AccountLogin;
