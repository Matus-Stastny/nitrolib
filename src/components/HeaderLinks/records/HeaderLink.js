// @flow strict
export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};
