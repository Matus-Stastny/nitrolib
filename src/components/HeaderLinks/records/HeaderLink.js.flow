// @flow strict
export type ActiveTab = "travel" | null;

export type HeaderLink = {|
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
|};

export type SearchForm = {|
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
|};
