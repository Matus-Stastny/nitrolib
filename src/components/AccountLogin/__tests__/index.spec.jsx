// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import AccountLogin from "..";
import { intlDefault } from "../../../records/Intl";

describe("#AccountLogin", () => {
  test("render", () => {
    const wrapper = shallow(
      <AccountLogin
        email=""
        onNoAccount={() => {}}
        onEmailChange={() => {}}
        onFacebookLogin={() => {}}
        onGoogleLogin={() => {}}
        onContinue={() => {}}
      />,
    );

    expect(wrapper.prop("children")(intlDefault)).toMatchSnapshot();
  });
});
