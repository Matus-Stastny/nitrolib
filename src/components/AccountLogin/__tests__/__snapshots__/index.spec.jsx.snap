// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AccountLogin render 1`] = `
<ForwardRef>
  <ModalSection>
    <Illustration
      name="Login"
      size="small"
      spaceAfter="small"
    />
    <Heading
      element="h2"
      spaceAfter="small"
    >
      <Translate
        html={false}
        t="account.manage_your_bookings"
        transform={[Function]}
        values={Object {}}
      />
    </Heading>
    <Text
      spaceAfter="large"
    >
      <Translate
        html={false}
        t="account.sign_in_description"
        transform={[Function]}
        values={Object {}}
      />
    </Text>
  </ModalSection>
  <ModalSection>
    <Text
      spaceAfter="medium"
      weight="bold"
    >
      <Translate
        html={false}
        t="account.sign_in_description"
        transform={[Function]}
        values={Object {}}
      />
    </Text>
    <Stack
      align="end"
      spaceAfter="small"
      spacing="condensed"
    >
      <ForwardRef
        label="account.email"
        onChange={[Function]}
        placeholder="account.email_placeholder"
        type="email"
        value=""
      />
      <Button
        onClick={[Function]}
      >
        <Translate
          html={false}
          t="account.continue"
          transform={[Function]}
          values={Object {}}
        />
      </Button>
    </Stack>
    <Text
      size="small"
    >
      <TextLink
        onClick={[Function]}
        type="secondary"
      >
        <Translate
          html={false}
          t="account.i_dont_have_account"
          transform={[Function]}
          values={Object {}}
        />
      </TextLink>
    </Text>
  </ModalSection>
  <ModalSection
    suppressed={true}
  >
    <Text
      spaceAfter="medium"
      weight="bold"
    >
      <Translate
        html={false}
        t="account.or_social_account"
        transform={[Function]}
        values={Object {}}
      />
    </Text>
    <Stack
      align="end"
      spacing="natural"
    >
      <Button
        block={true}
        bordered={true}
        icon={<Facebook />}
        onClick={[Function]}
        type="facebook"
      >
        <Translate
          html={false}
          t="account.log_in_with"
          transform={[Function]}
          values={
            Object {
              "provider": "Facebook",
            }
          }
        />
      </Button>
      <Button
        block={true}
        bordered={true}
        icon={<Google />}
        onClick={[Function]}
        type="google"
      >
        <Translate
          html={false}
          t="account.log_in_with"
          transform={[Function]}
          values={
            Object {
              "provider": "Google",
            }
          }
        />
      </Button>
    </Stack>
  </ModalSection>
</ForwardRef>
`;
